### PR TITLE
csv: handling of unparsable unicode

### PR DESCRIPTION
**Pull request by slint**
* BETTER Handle a wider variety files than what Python's csv module
  can support (e.g. unsupported Unicode characters and encodings).

Signed-off-by: Alexander Ioannidis <a.ioannidis@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@slint please provide minimal example file as we talked IRL. Thanks

File: invenio_previewer/extensions/zip.py [hunk 1]
```diff
@@ -40,10 +40,11 @@
 def make_tree(file):
     """Create tree structure from ZIP archive."""
     max_files_count = current_app.config.get('PREVIEWER_ZIP_MAX_FILES', 1000)
-    fp = file.open()
-    zf = zipfile.ZipFile(fp)
+
     tree = {'type': 'folder', 'id': -1, 'children': {}}
     try:
+        fp = file.open()
```
Comment (jirikuncar) [inline invenio_previewer/extensions/zip.py]: @slint shall we also use `with file.open() as fp:`?

File: invenio_previewer/templates/invenio_previewer/zip.html [hunk 2]
```diff
@@ -56,40 +56,49 @@
...
       <i class="fa fa-file-zip-o"></i> {{ file.filename }}
       <div id="fullScreenMode" title="{{ _('Full screen') }}" class="pull-right"><i class="fa fa-fw fa-arrows-alt"></i></div>
     </div>
   </div>
-  {% if- limit_reached %}
-  <div class="panel-heading panel-heading-alert">
-    <div class="panel-title">
-      <i class="fa fa-exclamation"></i>
-      {{ _('The previewer is not showing all the files') }}
+    {% if limit_reached %}
```
Comment (jirikuncar) [inline invenio_previewer/templates/invenio_previewer/zip.html]: please check the indentation here (-2 spaces?)

**Comment by jirikuncar** [comment]
@slint please amend a commit message too: `s/BETTER Handle/BETTER Handles/` Thanks!

**Comment by slint** [comment]
ping @jirikuncar

File: invenio_previewer/extensions/zip.py [hunk 3]
```diff
@@ -40,33 +42,45 @@
...
+        with file.open() as fp:
+            zf = zipfile.ZipFile(fp)
+            # Detect filenames encoding.
+            sample = ' '.join(zf.namelist()[:max_files_count])
+            if not isinstance(sample, binary_type):
+                sample = sample.encode('utf-16be')
+            encoding = chardet.detect(sample).get('encoding')
+            for i, info in enumerate(zf.infolist()):
+                if i > max_files_count:
+                    raise BufferError('Too much files inside the ZIP file')
```
Comment (jirikuncar) [inline invenio_previewer/extensions/zip.py]: much or many?

**Pull request merged**